tests/unittests/server/test_file_operations.py

```
@@ -0,0 +1,682 @@
+# Upload partial write cleanup
+# =============================================================================
+
+
+class TestPartialWriteCleanup:
+    """Failed uploads don't leave partial files behind."""
+
+    @pytest.mark.asyncio
+    async def test_no_file_left_on_stream_error(self, enable_file_ops):
+        """If the stream raises mid-upload, no partial file remains."""
+
+        async def failing_stream():
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `failing_stream`

(ANN202)